source/common/router/upstream_request.cc

```diff
@@ -106,6 +106,8 @@ UpstreamRequest::UpstreamRequest(RouterFilterInterface& parent,
     }
   }
   stream_info_.setUpstreamInfo(std::make_shared<StreamInfo::UpstreamInfoImpl>());
+  // route_ can be used to set route kind of metadata in custom tags.
```

Comment:
I don't think this comment is necessary, everything we're setting on the stream info is used for custom tags